core/src/components/radio/radio.tsx

```diff
@@ -24,7 +24,9 @@ import { createColorClasses, hostContext } from '../../utils/theme';
     ios: 'radio.ios.scss',
     md: 'radio.md.scss',
   },
-  shadow: true,
+  shadow: {
+    delegatesFocus: true,
```

Comment:
Threading:

This behavior does not work on Safari. Tabbing while focused on the first radio group moves focus to the `body`.


Also pressing the "Down" arrow selects the same radio.

Reply:
Also, does tabbing between radio groups work in Safari on `main` with the legacy syntax? If not, then I think a separate ticket to fix that would be fine since we aren't regressing behavior.

Reply:
It does work. However, `<ion-radio>.focus()` [still wouldn't work](https://github.com/ionic-team/ionic-framework/issues/27438). `delegatesFocus` was helping to fix that.
-
Tabbing on radio groups work on `main`. But it stops working if `delegatesFocus` is introduced.

Reply:
The PR description links to https://github.com/ionic-team/ionic-framework/issues/27268, but it sounds like we are also trying to fix https://github.com/ionic-team/ionic-framework/issues/27438. I think we are going to have the most luck with this PR if we focus on fixing one bug at a time.

Updating the internal `setFocus` method to focus the native `input` element should fix #27268 since the radio group calls `setFocus`. It sounds like the other issue is something that is specific to Firefox and may require additional investigation.